packages/earn-controller/src/EarnController.ts

```
@@ -0,0 +1,334 @@
+      this.#stakeSDK = null;
+      return;
+    }
+
+    const provider = new Web3Provider(networkClient.provider);
+    const { chainId } = networkClient.configuration;
+
+    // Initialize appropriate contracts based on chainId
+    const config: StakeSdkConfig = {
+      chainId: convertHexToDecimal(chainId),
+    };
+
+    try {
+      this.#stakeSDK = StakeSdk.create(config);
+      this.#stakeSDK.pooledStakingContract.connectSignerOrProvider(provider);
+    } catch (error) {
+      this.#stakeSDK = null;
+      // Only log unexpected errors, not unsupported chain errors
+      if (
+        !(
+          error instanceof Error &&
+          error.message.includes('Unsupported chainId')
+        )
+      ) {
+        console.error('Stake SDK initialization failed:', error);
+      }
+    }
+  }
+
+  #getCurrentAccount() {
+    return this.messagingSystem.call('AccountsController:getSelectedAccount');
+  }
+
+  #getCurrentChainId(): number {
+    const { selectedNetworkClientId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    const {
+      configuration: { chainId },
+    } = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      selectedNetworkClientId,
+    );
+    return convertHexToDecimal(chainId);
+  }
+
+  async refreshPooledStakingData(): Promise<void> {
+    const currentAccount = this.#getCurrentAccount();
+    if (!currentAccount?.address) {
+      return;
+    }
+
+    const chainId = this.#getCurrentChainId();
+    const apiService = this.#stakingApiService;
+
+    try {
+      const { accounts, exchangeRate } = await apiService.getPooledStakes(
+        [currentAccount.address],
+        chainId,
+      );
+
+      this.update((state) => {
+        state[EarnProductType.POOLED_STAKING].pooledStakes = accounts[0];
+        state[EarnProductType.POOLED_STAKING].exchangeRate = exchangeRate;
+      });
+    } catch (error) {
+      console.error('Failed to fetch pooled stakes:', error);
```

Rather than suppressing this error here what are your thoughts on doing this at the client level? Otherwise this error is just totally lost. (Same goes for the other methods here, really.)

---

I think you are right. I can have these methods throw errors so when we use this method on the client side we atleast have it come up on the client and handle them there as subscripiton methods are anyways logging these.